test/test_helper.rb

```diff
@@ -20,27 +20,35 @@
 require 'fileutils'
 FileUtils.mkdir_p(File.expand_path('../../tmp/cache', __FILE__))
 
+# https://github.com/seattlerb/minitest/blob/master/lib/minitest/autorun.rb
 gem 'minitest'
-require 'minitest/autorun'
-require 'minitest/reporters'
-Minitest::Reporters.use!
-if defined?(Minitest::Test)
-  $minitest_version = 5 # rubocop:disable Style/GlobalVars
-  # Minitest 5
-  # https://github.com/seattlerb/minitest/blob/e21fdda9d/lib/minitest/autorun.rb
-  # https://github.com/seattlerb/minitest/blob/e21fdda9d/lib/minitest.rb#L45-L59
-else
-  $minitest_version = 4 # rubocop:disable Style/GlobalVars
+begin
+  require 'minitest'
+rescue LoadError
+  # Minitest 4
+  require 'minitest/unit'
+  require 'minitest/spec'
+  require 'minitest/mock'
+  $minitest_version = 4
   # Minitest 4
   # https://github.com/seattlerb/minitest/blob/644a52fd0/lib/minitest/autorun.rb
   # https://github.com/seattlerb/minitest/blob/644a52fd0/lib/minitest/unit.rb#L768-L787
   # Ensure backward compatibility with Minitest 4
   Minitest = MiniTest unless defined?(Minitest)
   Minitest::Test = MiniTest::Unit::TestCase
-  def Minitest.after_run(&block)
-    MiniTest::Unit.after_tests(&block)
-  end
+  minitest_run = ->(argv) { MiniTest::Unit.new.run(argv) }
```

Comment:
basically copied from minitest autorun in versions 4 and 5. the tl;dr is that autorun is just sugar for `at_exit { MiniTest::Unit.new.run(ARGV) }` in 4 and `at_exit { Minitest.run(ARGV) }` in 5.  Copied because we're kind of writing our own 'autorun' here, by controlling when minitest starts in the `MiniTestHack` class below